lib/setupmanagers/qemuhck/devices/vhost-user-fs-pci.json

```diff
@@ -12,7 +12,8 @@
   "pre_start_commands": [
     "mkdir -p @fs_daemon_share_path@",
     "rm -f @fs_daemon_socket@",
-    "@fs_daemon_bin@ --socket-path=@fs_daemon_socket@ -o source=@fs_daemon_share_path@ -o cache=always &"
+    "@fs_daemon_bin@ --socket-path=@fs_daemon_socket@ -o source=@fs_daemon_share_path@ --cache always &",
+    "timeout 30 bash -c 'while [ ! -S \"@fs_daemon_socket@\" ]; do echo \"Socket does not exist, waiting...\"; sleep 1; done'"
```

Comment:
- Use `until`.
- It may be better to apply timeout for the entire initialization sequence on the Ruby side.